### PR TITLE
Update MarkovModel.py

### PR DESCRIPTION
**Pull request by Gasta88**
As mentioned in issue #1203, I've added public docstring to `Bio.MarkovModel` module.

### CODE REVIEW

File: Bio/MarkovModel.py [hunk 1]
```diff
@@ -43,6 +44,7 @@ def logaddexp(logx, logy):
 
 
 def itemindex(values):
+    """Return a dictionary that hold values and their position inside the sequence as key."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: hold --> holds.

Maybe `` """Return a dictionary of values with their sequence offset as keys."""`` is clearer?

File: Bio/MarkovModel.py [hunk 2]
```diff
@@ -75,14 +81,15 @@ def __str__(self):
...
+    """Read the first line and evaluate that begisn with the correct start (PRIVATE)."""
     line = handle.readline()
     if not line.startswith(start):
         raise ValueError("I expected %r but got %r" % (start, line))
     return line
 
 
 def load(handle):
-    """load(handle) -> MarkovModel()"""
+    """Load(handle) -> MarkovModel()."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: This is written like a function signature, not human readable. How about: ``"""Parse a file handle into a MarkovModel object."""``

File: Bio/MarkovModel.py [hunk 3]
```diff
@@ -140,25 +147,7 @@ def train_bw(states, alphabet, training_data,
...
-    optional parameters that you can use to assign pseudo-counts to
-    different matrices.  They should be matrices of the appropriate
-    size that contain numbers to add to each parameter matrix, before
-    normalization.
-
-    update_fn is an optional callback that takes parameters
-    (iteration, log_likelihood).  It is called once per iteration.
-
-    """
+    """Train a MarkovModel using the Baum-Welch algorithm."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: I would keep the existing three paragraphs of docstring information here.

File: Bio/MarkovModel.py [hunk 4]
```diff
@@ -388,23 +373,7 @@ def _backward(N, T, lp_transition, lp_emission, outputs):
...
-    from the alphabet, and observed states is a list of states from
-    states.
-
-    pseudo_initial, pseudo_transition, and pseudo_emission are
-    optional parameters that you can use to assign pseudo-counts to
-    different matrices.  They should be matrices of the appropriate
-    size that contain numbers to add to each parameter matrix
-
-    """
+    """Train a visible MarkovModel using maximum likelihoood estimates for each of the parameters."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: I would keep the existing three paragraphs of docstring information here.

File: Bio/MarkovModel.py [hunk 5]
```diff
@@ -479,11 +449,12 @@ def _mle(N, M, training_outputs, training_states, pseudo_initial,
 
 
 def _argmaxes(vector, allowance=None):
+    """Return indeces of the maximum values aong the vector (PRIVATE)."""
     return [numpy.argmax(vector)]
 
 
 def find_states(markov_model, output):
-    """find_states(markov_model, output) -> list of (states, score)"""
+    """Find_states(markov_model, output) -> list of (states, score)."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: Again this is essentially a function signature, can you rephrase it? eg

```python
def find_states(markov_model, output):
    """Find states in the given Markov model output.

    Returns a list of (states, score) tuples.
    """
```

File: Bio/MarkovModel.py (hunk 1 above)
Comment (Gasta88) [inline Bio/MarkovModel.py]: Better, I couldn't really find the right words without repeating myself.

File: Bio/MarkovModel.py [hunk 6]
```diff
@@ -267,9 +272,7 @@ def _baum_welch(N, M, training_outputs,
 def _baum_welch_one(N, M, outputs,
                     lp_initial, lp_transition, lp_emission,
                     lpseudo_initial, lpseudo_transition, lpseudo_emission):
-    # Do one iteration of Baum-Welch based on a sequence of output.
-    # NOTE: This will change the values of lp_initial, lp_transition,
-    # and lp_emission in place.
+    """Execute one step for Baum-Welch algorithm, changing the value for lp_initial, lp_transition and lp_emission in place (PRIVATE)."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: That's overly long. How about:

```python
"""Execute one step for Baum-Welch algorithm (PRIVATE).

Do one iteration of Baum-Welch based on a sequence of output.
Changes the value for lp_initial, lp_transition and lp_emission in place.
"""

File: Bio/MarkovModel.py [hunk 7]
```diff
@@ -347,10 +350,7 @@ def _baum_welch_one(N, M, outputs,
 
 
 def _forward(N, T, lp_initial, lp_transition, lp_emission, outputs):
-    # Implement the forward algorithm.  This actually calculates a
-    # Nx(T+1) matrix, where the last column is the total probability
-    # of the output.
-
+    """Implement forward algorithm (PRIVATE)."""
```
Comment (peterjc) [inline Bio/MarkovModel.py]: I think the original comment had useful information worth including as the main text of the new docstring.

**Comment by peterjc** [comment]
Thank you :)
